lib/miq_automation_engine/engine/miq_ae_service_model_base.rb

```diff
@@ -94,9 +96,9 @@ def self.expose(*args)
           return options[:override_return] if options.key?(:override_return)
           reflection = @object.class.reflection_with_virtual(method)
           if reflection && reflection.collection?
-            wrap_results(ret.to_a)
+            options[:association] ? wrap_results(self.class.filter_objects(ret.to_a)) : wrap_results(ret.to_a)
           else
-            wrap_results(ret)
+            options[:association] ? wrap_results(self.class.filter_objects(ret)) : wrap_results(ret)
```

Comment:
yea, let me try this again

Removing the `.to_a` going into rbac will reduce your # of queries significantly.
Of course, the `Array.wrap` that you have in there totally negates this savings. So you would need to remove that as well.